agent/verify_job.go

```diff
@@ -177,6 +177,41 @@ func (r *JobRunner) verifyJob(ctx context.Context, keySet any) error {
 			// So, we don't need to confirm that the values in the job are the same as those in the step.
 			continue
 
+		case "secrets":
+			jobSecrets := r.conf.Job.Step.Secrets
+
+			if len(step.Secrets) == 0 && len(jobSecrets) == 0 {
+				r.agentLogger.Debug("verifyJob: both job.Step.Secrets and step.Secrets are empty")
+				continue // both empty
+			}
+
+			if len(step.Secrets) != len(jobSecrets) {
+				r.agentLogger.Debug("failed to verifyJob: step.Secrets length %d != jobSecrets length %d", len(step.Secrets), len(jobSecrets))
+				return newInvalidSignatureError(ErrInvalidJob)
+			}
+
+			for i, stepSecret := range step.Secrets {
+				jobSecret := jobSecrets[i]
+
+				if stepSecret == nil && jobSecret == nil {
+					continue
+				}
```

Comment:
does this mean that yaml `null` is a valid secret? i think that's not necessarily a bad thing if so, but it is interesting.